src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,190 @@
+
+        try {
+            if (command.equals("list")) {
+                tasks.listTasks();
+            } else if (command.equals("delete")) {
+                if (inputArr.length == 1) {
+                    throw new NoTaskIDException();
+                } else {
+                    String strIndex = inputArr[1];
+                    if (isNumber(strIndex)) {
+                        int index = Integer.parseInt(strIndex) - 1; //because index starts from 1
+                        tasks.deleteTask(index, storage);
+                    } else {
+                        //case where a number was not provided
+                        throw new InvalidTaskIDException();
+                    }
+                }
+            } else if (command.equals("mark")) {
+                if (inputArr.length == 1) {
+                    throw new NoTaskIDException();
+                } else {
+                    String strIndex = inputArr[1];
+                    if (isNumber(strIndex)) {
+                        int index = Integer.parseInt(strIndex) - 1; //because index starts from 1
+                        if (tasks.isValidTaskID(index)) {
+                            tasks.markTask(index, storage);
+                        }
+                    } else {
+                        throw new InvalidTaskIDException();
+                    }
+                }
+            } else if (command.equals("unmark")) {
+                if (inputArr.length == 1) {
+                    throw new NoTaskIDException();
+                } else {
+                    String strIndex = inputArr[1];
+                    if (isNumber(strIndex)) {
+                        int index = Integer.parseInt(strIndex) - 1; //because index starts from 1
+                        if (tasks.isValidTaskID(index)) {
+                            tasks.unmarkTask(index, storage);
+                        }
+                    } else {
+                        throw new InvalidTaskIDException();
+                    }
+                }
+            } else if (command.equals("todo")) {
+                parseToDo(input);
+            } else if (command.equals("deadline")) {
+                parseDeadline(input);
+            } else if (command.equals("event")) {
+                parseEvent(input);
+            } else {
+                throw new InvalidCommandException();
+            }
+        } catch (DukeException e) {
+            ui.showError(e.getMessage());
+        }
+    }
+
+    public void parseToDo(String input) throws DukeException {
+        String[] inputArr = input.split(" ", 2);
+        if (inputArr.length == 1) {
+            throw new NoDescException();
+        } else {
+            if (inputArr[1].isBlank()) {
+                throw new NoDescException();
+            } else {
+                //0 for unmarked, any other number for marked
+                ToDo toDo = new ToDo(0, inputArr[1]);
+                tasks.addTask(toDo, storage);
```

Comment:
Perhaps acting on the inputs can be put in another class? IMO the name parser doesn't really indicate that this behaviour is expected.